ophyd/areadetector/plugins.py

```diff
@@ -632,6 +633,56 @@ class ROIPlugin(PluginBase):
                   doc='Minimum size of the ROI in XYZ',
                   default_read_attrs=('min_x', 'min_y', 'min_z'))
 
+    def set(self, region):
+        ''' This functions allows for the ROI regions to be set.
+
+        This function takes in an ROI_number, and a dictionary of tuples and sets the ROI region.
+
+        PARAMETERS
+        ----------
+        region: dictionary.
+            A dictionary defining the region to be set, which has the structure:
+            { 'x':[min, size], 'y':[min, size], 'z':[min, size]}. 
+            Any of the keywords can be ommitted, and they will be ignored.
+        '''
+        if region is not None:
+            for direction, value in region.items():
+                getattr(self, f'min_xyz.min_{direction}').put(value[0])
+                getattr(self, f'size.{direction}').put(value[1])
+
+        return NullStatus()
+        
+    def read(self):
```

Comment:
Could this be done by setting the right `read_attrs` on the ROI and the right names on the children?

Reply:
I am not sure on this, possibly but the issue is that the children that are referenced here in read are a few levels down the tree and I was hoping to avoid including this level of detail in the read.

Reply:
Would setting

```
_default_read_attrs = ['x', 'y', 'z', 'status']
```

and using the standard `read()` method achieve the same effect as this custom `read()` and `describe()` methods?

Reply:
I am not sure it would because 'x' is actually the attribute  min_xyz.min_x so I would need to use
_default_read_attrs = ['min_xyz_min_x', .........
which is what I was trying to avoid with the custom read as then the read is a little to verbose for my liking?

also 'status' in this case would need to be named 'enable' (which I think is OK).

Reply:
> the read is a little to verbose for my liking

Meaning the keys in the dictionary returned by read? The hook that ophyd provides for fixing that is in changing the attribute's `.name`.

> Also I would have to also include the 'size' attributes seperately where as in my read method I combine them into a tuple.

This kind of rearrangement should be done by making special Signals that give you the readings you want. The generality of `read()` across all devices is a really powerful assumption to be able to rely on, one we shouldn't break unless we have no other option, because the cost of having to account for specialize `read()` behavior in downstream code is high.